src/digraph.jl

```diff
@@ -76,22 +76,23 @@ end
 
 is_directed(g::DiGraph) = true
 
-function unsafe_add_edge!(g::DiGraph, e::Edge)
-    push!(g.fadjlist[src(e)], dst(e))
-    push!(g.badjlist[dst(e)], src(e))
+function add_edge!(g::DiGraph, e::Edge)
+    s, d = e
+    s in vertices(g) || error("Vertex $s not in graph")
+    d in vertices(g) || error("Vertex $d not in graph")
+    _insert_and_dedup!(g.fadjlist[s], d)
+    _insert_and_dedup!(g.badjlist[d], s)
     push!(g.edges, e)
     return e
 end
 
 
 function rem_edge!(g::DiGraph, e::Edge)
-    reve = reverse(e)
     has_edge(g,e) || error("Edge $e is not in graph")
```

Comment:
this is ok for now but once we get rid of the edge this  can be eliminated  in favour of ```i != 0 || error("Edge $e is not in graph")``` a few lines later